### ISSUE TITLE
Kernel eBPF problems.

### ISSUE THREAD

**Issue by sergeykostov** [vmware/photon#1233]
Hello, 
there seems to be some miss-configuration of the eBPF in the kernel.
For example try to run all bpftrace tools that come with the official package of PHOTON, 
you will notice most of them failing or not giving any output.

Please activate the following configuration options in the kernel also.
CONFIG_CGROUP_BPF=y
CONFIG_BPFILTER=y
CONFIG_BPFILTER_UMH=m
CONFIG_NET_ACT_BPF=m

**Comment by ssahani** [comment]
Which version of photon is this ?

**Comment by ssahani** [comment]
yes it's not enabled. we need to enable this. Markging as RFE https://github.com/vmware/photon/blob/master/SPECS/linux/config_x86_64#L172

**Comment by sergeykostov** [comment]
All 3 versions. However I think it is also present in the 4 series.

**Comment by akdanz** [comment]
There is more. We are using the official Photon OS 5.0 OVA image with latest esx kernel (6.1.130-1.ph5-esx) and it's lacking the following config options for proper compatibiliy with Crowdstrike Falcon sensor:

```
CONFIG_DEBUG_INFO_BTF=y
CONFIG_BPF_EVENTS=y
CONFIG_KALLSYMS_ALL=y
CONFIG_BSD_PROCESS_ACCT=y
```

**Comment by akdanz** [comment]
During the past weeks I went through that already, together with Crowdstrike support agents. It all boils down to the missing kernel config items that I mentioned.

According to the official docs, they require all of the following:
```
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
CONFIG_DEBUG_INFO_BTF=y
CONFIG_TRACING=y
CONFIG_KPROBE_EVENTS=y
CONFIG_UPROBE_EVENTS=y
CONFIG_BPF_JIT=y
CONFIG_SECURITY=y
CONFIG_KALLSYMS_ALL=y
CONFIG_PROC_FS=y
CONFIG_BSD_PROCESS_ACCT=y
CONFIG_CGROUPS=y
CONFIG_CGROUP_BPF=y

CONFIG_NET_CLS_BPF=y or m
CONFIG_NET_CLS_ACT=y or m
CONFIG_NET_SCH_INGRESS=y or m

CONFIG_DEBUG_INFO_BTF_MODULES=y
```

For reference, see (requires a login): https://falcon.crowdstrike.com/documentation/page/edd7717e/falcon-sensor-for-linux-system-requirements#ef1f1492 and https://supportportal.crowdstrike.com/s/article/Release-Notes-Falcon-Sensor-for-Linux-User-Mode

**Comment by akaher** [comment]
All the following config enabled in generic kernel:

```
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
CONFIG_DEBUG_INFO_BTF=y
CONFIG_TRACING=y
CONFIG_KPROBE_EVENTS=y
CONFIG_UPROBE_EVENTS=y
CONFIG_BPF_JIT=y
CONFIG_SECURITY=y
CONFIG_KALLSYMS_ALL=y
CONFIG_PROC_FS=y
CONFIG_BSD_PROCESS_ACCT=y
CONFIG_CGROUPS=y
CONFIG_CGROUP_BPF=y

CONFIG_NET_CLS_BPF=y or m
CONFIG_NET_CLS_ACT=y or m
CONFIG_NET_SCH_INGRESS=y or m

CONFIG_DEBUG_INFO_BTF_MODULES=y
```

Please use generic kernel i.e. linux not linux-esx.
Execute following command to install linux generic  flavour: 
    tdnf install linux

-Ajay

**Comment by akdanz** [comment]
We are running the official OVL image on VSphere / ESX which ships with esx kernel by default because it's "optimized" for that use case.

Will there be any downsides from switching the kernel?

**Comment by akaher** [comment]
linux-esx is preferable to use on ESXi, on other side on harm to use linux generic on ESX.
As linux-esx doesn't fulfill your requirements, so you should use linux generic.
